CHANGELOG.md

```diff
@@ -26,6 +26,7 @@
 - PR #5971 Add cuStreamz README for basic installation and use
 - PR #6024 Expose selecting multiple ORC stripes to read from Python
 - PR #6155 Use the CUB submodule in Thrust instead of fetching CUB separately
+- PR #6321 Switch JNI to use `arena_memory_resource`
```

Comment:
Nit: We should update the changelog since we're no longer forcing the arena memory resource but just adding support for requesting it.